girder/utility/gridfs_assetstore_adapter.py

```diff
@@ -46,13 +47,49 @@
 def _ensureChunkIndices(collection):
     """
     Ensure that we have appropriate indices on the chunk collection.
+
+    :param collection: a connection to a mongo collection.
     """
     collection.create_index([
         ('uuid', pymongo.ASCENDING),
         ('n', pymongo.ASCENDING)
     ], unique=True)
 
 
+def _setupSharding(collection, keyname='uuid'):
```

Comment:
The default keyname is 'uuid', but in the adapter it's called 'chunkUuid', and you are using the default value when this is called.